packages/router/src/router.js

```diff
@@ -38,10 +38,26 @@ export function RouterProvider( { children } ) {
 	);
 
 	useEffect( () => {
+		// `/wp_template_part/all` path is no longer used and redirects to
+		// Patterns page for backward compatibility.
+		const { path } = location?.params;
+
+		if ( path === '/wp_template_part/all' ) {
+			history.push( {
+				path: '/patterns',
+			} );
+			setLocation( {
+				...location,
+				params: {
+					path: '/patterns',
+				},
+			} );
```

Comment:
Only `history.push` was needed here, the location would be updated automatically by the `history.listen` listener.